### ISSUE TITLE
Launcher crashes on Fedora 36

### ISSUE THREAD

**Issue by IsraPerez98** [kyleraykbs/Portal2-32PlayerMod#27]
The .sh file crashes with the following error:

> pygame 2.1.2 (SDL 2.0.16, Python 3.10.4)
> Hello from the pygame community. https://www.pygame.org/contribute.html
> libGL error: MESA-LOADER: failed to open radeonsi: /tmp/_MEIKSjnpG/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib64/dri/radeonsi_dri.so) (search paths /usr/lib64/dri, suffix _dri)
> libGL error: failed to load driver: radeonsi
> libGL error: MESA-LOADER: failed to open swrast: /tmp/_MEIKSjnpG/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib64/dri/swrast_dri.so) (search paths /usr/lib64/dri, suffix _dri)
> libGL error: failed to load driver: swrast
> X Error of failed request:  BadValue (integer parameter out of range for operation)
>   Major opcode of failed request:  150 (GLX)
>   Minor opcode of failed request:  3 (X_GLXCreateContext)
>   Value in failed request:  0x0
>   Serial number of failed request:  105
>   Current serial number in output stream:  106

**Comment by cabiste69** [comment]
what linux distro are you using?

**Comment by cabiste69** [comment]
well done a google search and it's a GPU related issue so we can't do anything about it
just make sure that your drivers are up to date

**Comment by KonradCzerw** [comment]
One possible solution may be to run the launcher manually instead of using the pre-built version, since from what I've managed to find it may also be related to incompatibilities between the system `libstdc++.so` and the one embedded into the package. This issue is also related to version de-synchronizations of libraries between Linux distros, which appimages and appimage like formats (including pyinstallers' one) attemt to fix but fail at horribly.

TL;DR try running from source and python packers suck

**Comment by IsraPerez98** [comment]
> One possible solution may be to run the launcher manually instead of using the pre-built version, since from what I've managed to find it may also be related to incompatibilities between the system `libstdc++.so` and the one embedded into the package. This issue is also related to version de-synchronizations of libraries between Linux distros, which appimages and appimage like formats (including pyinstallers' one) attempt to fix but fail at horribly.
> 
> TL;DR try running from source and python packers suck

Running from source does indeed fix the issue and the game launches without any problems, also i'm not sure what framework you're using, but i feel like you should add a "requirements.txt" file for python.

Anyways great project, I've been having a lot of fun with this mod.
